File: 20250408/leetcode3396. 使数组元素互不相同所需的最少操作次数/main.cpp
#include <cmath>
#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
  int minimumOperations(vector<int> &nums) {
    int n = nums.size();
    unordered_set<int> set;
    for (int i = n - 1; i >= 0; i--) {
      if (set.count(nums[i])) {
        return int(ceil((i + 1) / 3.0));
      }
      set.insert(nums[i]);
    }
    return 0;
  }
};

File: 20250408/leetcode3396. 使数组元素互不相同所需的最少操作次数/main.js
// 给你一个整数数组 nums，你需要确保数组中的元素 互不相同 。为此，你可以执行以下操作任意次：

// 从数组的开头移除 3 个元素。如果数组中元素少于 3 个，则移除所有剩余元素。
// 注意：空数组也视作为数组元素互不相同。返回使数组元素互不相同所需的 最少操作次数 。

/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumOperations = function (nums) {
  const set = new Set();
  const n = nums.length;
  for (let i = n - 1; i >= 0; i--) {
    if (set.has(nums[i])) {
      return Math.ceil((i + 1) / 3);
    }
    set.add(nums[i]);
  }
  return 0;
};
